gm_dotnet_managed/Tests/PushCFuncPtr.cs

```diff
@@ -0,0 +1,75 @@
+﻿using GmodNET.API;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace Tests
+{
+    // Tests ILua.PushCFunction(delegate* unmanaged[Cdecl]<IntPtr, int>)
+    public class PushCFuncPtr : ITest
+    {
+        static int random_number;
+
+        static string random_string;
+
+        static PushCFuncPtr()
+        {
+            random_number = new Random().Next(1, 500);
+
+            random_string = Guid.NewGuid().ToString();
+        }
+
+        [UnmanagedCallersOnly(CallConvs = new Type[] { typeof(CallConvCdecl) })]
+        static int TestFunc(IntPtr lua_state)
+        {
+            ILua lua = GmodInterop.GetLuaFromState(lua_state);
+
+            lua.PushNumber(random_number);
+
+            lua.PushString(random_string);
+
+            return 2;
+        }
+
+        public Task<bool> Start(ILua lua, GetILuaFromLuaStatePointer lua_extructor, ModuleAssemblyLoadContext assembly_context)
+        {
+            TaskCompletionSource<bool> taskCompletion = new();
+
+            try
+            {
+                unsafe
+                {
+                    lua.PushCFunction((delegate* unmanaged[Cdecl]<IntPtr, int>)(delegate* managed<IntPtr, int>)&TestFunc);
```

Comment:
This horrible double type cast will go away as soon as .NET 5 SDK RC2 is available.

Reply:
Will fix as soon as RC2 is out, works for now so we can merge as it is.

Reply:
It is a test, so there will be no changes to the core codebase anyway.